lib/ublksrv_cmd.c

```diff
@@ -117,15 +117,15 @@ static int __ublksrv_ctrl_cmd(struct ublksrv_ctrl_dev *dev,
 
 	ublksrv_ctrl_init_cmd(dev, sqe, data);
 
-	ret = io_uring_submit(&dev->ring);
+	do {
+		ret = io_uring_submit_and_wait(&dev->ring, 1);
+	} while (ret == -EINTR);
 	if (ret < 0) {
-		fprintf(stderr, "uring submit ret %d\n", ret);
+		fprintf(stderr, "uring submit_and_wait ret %d\n", ret);
 		return ret;
 	}
 
-	do {
-		ret = io_uring_wait_cqe(&dev->ring, &cqe);
-	} while (ret == -EINTR);
+	ret = io_uring_wait_cqe(&dev->ring, &cqe);
```

Comment:
Here it is still io_uring_wait_cqe(), :-)

If it is wait_cqe, you may have to deal with -EINTR too, that is why I'd suggest to take io_uring_peek_cqe().

Thanks,

Reply:
Whoops, my mistake. I forgot to amend my changes. The commit uses io_uring_peek_cqe() now